homeassistant/components/minecraft_server/__init__.py

```diff
@@ -0,0 +1,452 @@
+            self._players_max = status_response.players.max
+            self._latency = status_response.latency
+
+            if status_response.players.sample is None:
+                self._players_list = "[]"
+            else:
+                self._players_list = "["
+
+                for player in status_response.players.sample:
+                    self._players_list += player.name + ", "
+
+                # Remove last seperator ", " and add end bracket.
+                self._players_list = self._players_list[:-2] + "]"
+
+                # Limit players list length to 255.
+                if len(self._players_list) > 255:
+                    self._players_list = self._players_list[:-4] + "...]"
+                    _LOGGER.debug("Players list length > 255 (truncated).")
+
+    def _print_data(self):
+        """Print properties."""
+        _LOGGER.debug(
+            """Properties of '%s' (%s:%s):
+            %s: %s
+            %s: %s
+            %s: %s
+            %s: %s
+            %s: %s
+            %s: %s
+            %s: %s""",
+            self._name,
+            self._host,
+            self._port,
+            NAME_DESCRIPTION,
+            self._description,
+            NAME_VERSION,
+            self._version,
+            NAME_PROTOCOL_VERSION,
+            self._protocol_version,
+            NAME_LATENCY_TIME,
+            self._latency,
+            NAME_PLAYERS_ONLINE,
+            self._players_online,
+            NAME_PLAYERS_MAX,
+            self._players_max,
+            NAME_PLAYERS_LIST,
+            self._players_list,
+        )
+
+
+class MinecraftServerEntity(Entity):
+    """Representation of a Minecraft Server base entity."""
+
+    def __init__(self, hass, server, name, unit, icon):
+        """Initialize base entity."""
+        self._server = server
+        self._hass = hass
+        self._state = None
+        self._name = server.name() + " " + name
+        self._sensor_name = name
+        self._unit = unit
+        self._icon = icon
+        self._unique_id = f"{self._server.unique_id()}-{self._sensor_name}"
+        self._device_info = {
+            "identifiers": {(DOMAIN, self._server.unique_id())},
+            "name": self._server.name(),
+            "manufacturer": MANUFACTURER,
+            "model": f"Minecraft Server ({self._server.version()})",
+            "sw_version": self._server.protocol_version(),
+        }
+
+        # Subscribe to signal from server instance.
+        self._signal_name = self._server.signal_name()
+        async_dispatcher_connect(
+            self._hass, self._signal_name, self._async_trigger_update
+        )
+
+    async def _async_trigger_update(self):
+        """Triggers update of properties after receiving signal from server instance."""
+        _LOGGER.debug("%s: Received signal '%s'.", self._sensor_name, self._signal_name)
+        self.async_schedule_update_ha_state(force_refresh=True)
```

Comment:
Is it OK like this or should I even try to get rid of _async_trigger_update()? Also I'm still using async_schedule_update_ha_state() instead of async_write_ha_state(), as I didn't find any doc about it.

Reply:
I think you should remove `_async_update_callback` and still use `async_write_ha_state`, and test if working.

Reply:
Just tried it with `async_write_ha_state`, with no success. The states are not updated anymore. Also I think that `async_schedule_update_ha_state` is the preferred way. The docstring says: 

> Scheduling the update avoids executor deadlocks. [...] If state is changed more than once before the ha state change task hasbeen executed, the intermediate state transitions will be missed.